File: python/src/pkg/handler.py
from ddtrace import tracer

@tracer.wrap("first_function")
def first_function():
    second_function()

@tracer.wrap("second_function")
def second_function():
    try:
        third_function()
    except ValueError as e:
        raise ValueError('error when calling third_function in second_function') from e

@tracer.wrap("third_function")
def third_function():
    try:
        fourth_function()
    except ValueError as e:
        raise ValueError('error when calling fourth_function in third_function')

@tracer.wrap("fourth_function")
def fourth_function():
    try:
        s = "baba"
        a = s.split("a")
        print(a[5])
    except IndexError as e:
        raise ValueError('failed in fourth_function')

File: golang/src/pkg/handler.go
package pkg

import (
	"context"
	"errors"

	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
)

func FirstFunction(ctx context.Context) error {
	span, ctx := tracer.StartSpanFromContext(ctx, "first_function")
	defer span.Finish()

	if err := SecondFunction(ctx); err != nil {
		span.SetTag("error", err)
	}

	return nil
}

func SecondFunction(ctx context.Context) error {
	span, ctx := tracer.StartSpanFromContext(ctx, "second_function")
	defer span.Finish()

	if err := ThirdFunction(ctx); err != nil {
		err := errors.New("error when calling third_function in second_function")
		span.SetTag("error", err)
		return err
	}

	return nil
}

func ThirdFunction(ctx context.Context) error {
	span, ctx := tracer.StartSpanFromContext(ctx, "third_function")
	defer span.Finish()

	if err := FourthFunction(ctx); err != nil {
		err := errors.New("error when calling fourth_function in third_function")
		span.SetTag("error", err)
		return err
	}

	return nil
}

func FourthFunction(ctx context.Context) error {
	span, _ := tracer.StartSpanFromContext(ctx, "fourth_function")
	defer span.Finish()

	v := map[string]interface{}{}
	if _, ok := v["a"]; !ok {
		err := errors.New("failed in fourth_function")
		span.SetTag("error", err)
		return err
	}

	return nil
}
